halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/EchoService.java

```diff
@@ -68,6 +76,23 @@ public Type getType() {
     return profiles;
   }
 
+  protected Optional<Profile> generateSlackProfile(DeploymentConfiguration deploymentConfiguration, SpinnakerRuntimeSettings endpoints, String spinnakerHome) {
+    SlackNotification slackNotification = deploymentConfiguration.getNotifications().getSlack();
+    if (slackNotification.isEnabled()
```

Comment:
This type of logic should probably live in the profile factory. There is an exception to this for clouddriver + aws (where I'm guessing you saw this pattern) due to the fact that different deployment environments handle AWS differently.